danceradvent/public/articles/2015/7-medium-scale-part-5.pod

```diff
@@ -0,0 +1,208 @@
+=encoding utf8
+
+=head1 Medium-Scale Dancer, Part 5: REST APIs and JSON
+
+In previous parts of L<this article series|/2015/3>, I defined each
+route handler as a one-liner, like so:
+
+ prefix '/mf' => sub {
+     get '/' => \&retrieve;
+     prefix '/subfeature' => sub {
+         get  '/'    => \&_get;
+         post '/'    => \&_post;
+         put  '/'    => \&_put;
+         del  '/:id' => \&_del;
+     };
+ };
+
+This is very much on purpose, despite the fact that Dancer lets you
+define all of the code for each route handler within an anonymous code
+reference (e.g. C<sub { ... }>) block. What do we get by moving those
+functions elsewhere, and what do we gain that makes it worth tolerating
+the ugly C<\&> syntax?
+
+Simple: it makes the URL structure of the web app explicit in the code.
```

Comment:
There is another important gain, and where to put this in the article is up to you: When you use a named subroutine, instead of an anonymous one, the debugging becomes visibly easier in stack traces.

(However, you can drop a hint that named routes are coming in a future version of Dancer2!)